src/main/java/com/derongan/minecraft/mineinabyss/ascension/AscensionListener.kt

```diff
@@ -35,15 +39,45 @@ class AscensionListener : Listener {
         val changeY = to.y - from.y
         val playerData = player.playerData
 
-        if (playerData.isAffectedByCurse) {
+        if (player.isFlying) {
+            playerData.distanceAscended = 0.0
+        } else if (playerData.isAffectedByCurse) {
+            val section = worldManager!!.getSectionFor(to) ?: return
+            val layer = manager.getLayerForSection(section)
+            var curseFactor = getCurseStrength(manager, to)
+
             val dist = playerData.distanceAscended
-            playerData.distanceAscended = (dist + changeY).coerceAtLeast(0.0)
+            playerData.distanceAscended = (dist + curseFactor * changeY).coerceAtLeast(0.0)
```

Comment:
Maybe rename this variable. Right now with this impl its kind of misleading since its not actually distanceAscended.

Reply:
renamed to curseAccrued